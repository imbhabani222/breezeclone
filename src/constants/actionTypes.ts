import keyMirror from "keymirror";

const actionTypes = {
  FETCH_TODO_REQUEST: null,
  FETCH_TODO_SUCCESS: null,
  FETCH_TODO_FAILURE: null,
  SAMPLE_ACTION: null,
  ACTIONS: null,
  FETCH_IDENTITY: null,
  FETCH_IDENTITY_SUCCESS: null,
  FETCH_IDENTITY_FAILURE: null,
  FETCH_TEST_DETAILS: null,
  FETCH_TEST_DETAILS_SUCCESS: null,
  FETCH_TEST_DETAILS_FAILURE: null,
  FETCH_SERVER_TIME_REQUEST: null,
  FETCH_SERVER_TIME_SUCCESS: null,
  FETCH_SERVER_TIME_FAILURE: null,

  PROBLEM_DETAILS_REQUEST: null,
  PROBLEM_DETAILS_SUCCESS: null,
  PROBLEM_DETAILS_FAILURE: null,

  FETCH_QUESTIONS_REQUEST: null,
  FETCH_QUESTIONS_SUCCESS: null,
  FETCH_QUESTIONS_FAILURE: null,

  DELETE_ANSWER_REQUEST: null,
  DELETE_ANSWER_SUCCESS: null,
  DELETE_ANSWER_FAILURE: null,

  DELETE_FIB_REQUEST: null,
  DELETE_FIB_SUCCESS: null,
  DELETE_FIB_FAILURE: null,

  FETCH_PROBLEM_REQUEST: null,
  FETCH_PROBLEM_SUCCESS: null,
  FETCH_PROBLEM_FAILURE: null,

  POST_TEST_REQUEST: null,
  POST_TEST_SUCCESS: null,
  POST_TEST_FAILURE: null,

  FETCH_SINGLE_PRB_SOL_REQUEST: null,
  FETCH_SINGLE_PRB_SOL_SUCCESS: null,
  FETCH_SINGLE_MCQ_SOL_FAILURE: null,

  POST_SINGLE_PRB_SOL_REQUEST: null,
  POST_SINGLE_PRB_SOL_SUCCESS: null,
  POST_SINGLE_PRB_SOL_FAILURE: null,

  POST_FEEDBACk_REQUEST: null,
  POST_FEEDBACK_SUCCESS: null,
  POST_FEEDBACK_FAILURE: null,

  FETCH_SOLUTIONSET_REQUEST: null,
  FETCH_SOLUTIONSET_SUCCESS: null,
  FETCH_SOLUTIONSET_FAILURE: null,

  PATCH_FIB_REQUEST: null,
  PATCH_FIB_REQUEST_SUCCESS: null,
  PATCH_FIB_REQUEST_FAILURE: null,

  POST_REPORT_ISSUE_REQUEST: null,
  POST_REPORT_ISSUE_SUCCESS: null,
  POST_REPORT_ISSUE_FAILURE: null,

  PATCH_ANSWER_REQUEST: null,
  PATCH_ANSWER_REQUEST_SUCCESS: null,
  PATCH_ANSWER_REQUEST_FAILURE: null,

  POST_UPLOAD_FILE_REQUEST: null,
  POST_UPLOAD_FILE_SUCCESS: null,
  POST_UPLOAD_FILE_FAILURE: null,

  POST_SECTION_REQUEST: null,
  POST_SECTION_SUCCESS: null,
  POST_SECTION_FAILURE: null,

  GET_LANGUAGE_REQUEST: null,
  GET_LANGUAGE_SUCCESS: null,
  GET_LANGUAGE_FAILURE: null,

  RUN_CODE_CUSTOM_INPUT_REQUEST: null,
  RUN_CODE_CUSTOM_INPUT_SUCCESS: null,
  RUN_CODE_CUSTOM_INPUT_FAILURE: null,

  RUN_CODE_CUSTOM_INPUT_RESULT: null,

  TESTCASES_RUN_CODE__REQUEST: null,
  TESTCASES_RUN_CODE__SUCCESS: null,
  TESTCASES_RUN_CODE__FAILURE: null,
};

export default keyMirror(actionTypes);
