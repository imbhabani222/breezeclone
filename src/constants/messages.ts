const messages = {
  NOT_AUTHORIZED:
    "You are not authorized or session has been expired. Please try again.",
  NO_DATA_AVAILABLE: "No data available.",
  ACCESS_DENIED: "Access Denied/Forbidden.",
  DEFAULT_ERROR_MESSAGE:
    "There seems to be a problem, please retry after sometime.",
  USER_NOT_FOUND: "Invalid user",
  IMAGE_NOT_AVAILABLE: "Image not available",
  ARE_YOU_SURE: "Are you sure?",
  TYPE_OUTLINE: "outline",
  TYPE_RECTFILLED: "rectFilled",
  TYPE_FILLED: "filled",
  PLEASE_TYPE_ANSWER: "Please type your answer.",
  EXAM_NAME: "[0-2 years] Data Analytics",
  EXAM_TYPE: "By DoSelect UI/UX Testing",
  MODAL_CONTENT:
    "# Are you sure ? \n You still have **%hours% hours %minutes% mins %seconds% seconds** left to submit this assessment. </br></br>You have attempted %completed% of %total% questions.</br></br>You will not be able to attempt any question or make changes to your solution once you submit.",
  LOADING: "Loading...",
  PROCEED_TO_FEEDBACK: "PROCEED TO FEEDBACK PAGE",
  GOING_GREAT:
    "Going great .</br></br>You have attempted %completed% of %total% questions.",
  AUTO_NEXT_MODAL:
    "This section has been auto-submitted as the allotted time is up.</br></br>You have attempted %completed% of %total% questions.",
  BROWSER_TOLERANCE_INFO:
    "  You had navigated away from the test screen. This incident will be reported to the test administrator.",
  LOAD_MORE: "Load More",
  BACKBUTTON: " Go back to Assessment",
  PLATFORM_SUPPORT: "Platform Support",
  CLIENTS_SUPPORT: "Clients Support",
  HELP: "Help",
  QUESTION: "View All Questions",
  RIGHT: "right",
  FEEDBACK: {
    FEEDBACKCOMPLETED: "Woot! You've completed this assessment.",
    FEEDBACKTITLE: "SHARE FEEDBACK",
    FEEDBACKNOTIFIED:
      "We've notified team %orgName% about this. If you have any queries regarding this, please get in touch with your point of contact.",
    SENDFEEDBACK: "Send Feedback",
    ENTERFEWWORDS: "Please enter few words",
    MESSAGE_LABEL: "message",
    SUBMIT_LABEL: "Submit",
    FEEDBACKFORM: "feedbackForm",
    LIKETOTELL: "Is there something that you'd like to tell us?",
  },
  REPORT_LABLE: "REPORT AN ISSUE",
  REPORT_MODAL_TITLE: "Report an issue with Stack Using Interface",
  REPORT_MODAL_INPUT_LABEL: "  Other:",
  PROGRESS_DATABAR_TEXT: "%completed%/%total% Questions attempted",
  RUN_CODE_LABEL: "Run Code",
  ABORT_LABEL: "Abort",
  TEST_RESULTS_LABEL: "Test results",
  TEST_RESULTS_TEXT: 'Click on "Run code" to evaluate your solution',
  CUSTOM_INPUT_LABEL: "Custom Input",
  CUSTOM_INPUT_TEXT: "STDIN",
  ICON: "Icon",
  TIME_LABEL: {
    H: "H:",
    M: "M:",
    S: "S",
  },
  FAQ: {
    KEY1: "1",
    KEY2: "2",
    KEY3: "3",
    TAB1: "Help",
    TAB2: "Coding Environment",
    TAB3: "Status Code",
    SEARCH_PLACEHOLDER: "Ask a question",
    FAQINFOTEXT: `Please find below the general instructions and information about 
    DoSelect's assessment environment. We highly recommend that you go
    through this before you start taking the assessment.`,
  },
  TESTSTATUS: {
    ICONSTATUS: "iconsStatus",
    ACTIVATEICON: "activateIcon",
    DEACTIVATEICON: "deactivateIcon",
  },
  DESCRIPTION_LABEL: "Description",
  STATUS_CODE_LABEL: "Status Code",
  LIBRARIES_AVAILABLE_LABEL: "Libraries Available",
  TIME_LIMIT_LABEL: "Time Limit(in seconds)",
  MEMORY_UNIT_LABEL: "Memory Limit (in MB)",

  LIBRARIES_AVAILABLE_KEY: "Libraries available",
  TIME_LIMIT_KEY: "Time limit (in seconds)",
  MEMORY_UNIT_KEY: "Memory limit (in MB)",

  VERSION_LABEL: "Version",
  LANGUAGE_LABEL: "Language",
  DO_SELECT_RESEVE_RIGHTS: "Doselect. All rights reserved",
  LOGO_LABEL: "Logo",
  TEXT_LABEL: "text",
  TIME_FOR_EXAM_LABEL: "%hours% hours %minutes% mins",
  TIME_FOR_MODAL: "%hours% hours %minutes% mins %seconds% seconds",
  ALT_IMG: "icon Doselect",
  CANCEL: "Cancel",
  SUBMIT_END: "Submit and end the Assessment",
  SUBMIT_MOVE_NEXT: "Submit and Move to Next Section",
  PROCEED_TO_NEXT: "Proceed to next section",
  UPLOAD: "Upload",
  SAVE: "Save",
  QUESTION_TYPE: "MCQ + Coding + Fill in the  Blanks",
  LOGO: "do select logo",
  CAMERA: "Video Proctoring",
  NETWORK: "Network quality",
  TIMER: "Timer",

  BROWSER_FULLSCREEN: `<p>This %type% requires your browser window to be in fullscreen mode</p>`,
  UPLOAD_ATTACHMENT: "Upload an attachment with this solution.",
  VIDEO_TEST: {
    DESC: "Description",
    VID_DUR: "Video Duration",
    SEC: "seconds",
    RECORD: "Record",
    STOP: "Stop",
    RESUME: "Resume",
    PAUSE: "Pause",
    SUBMIT: "Submit",
    REC_DUR: "Record duration:",
  },

  CODING: {
    CODING: "Coding",
    RELOAD_STUB: "Reload Code Stub?",
    RELOAD_STUB_CONTENT: `You have changed the technology of this solution. Do you want
    to REPLACE your current code with the default code stub of
    this technology? Please note that you will lose your current
    code by doing this.`,
    NO_TEST_FOUND: "No Test Cases Found",
    RUN_ALL_TEST_CASES: "Run all Test Cases",
    DESCRIPTION: "Description",
    DESCRIPTIONS: "Descriptions",
    INPUT: "Input",
    OUTPUT: "Output",
    EXPECTED_OUTPUT: "Expected Output",
    EVAL_TIME: "Eval Time",
    CPU: "CPU",
    MEMORY: "Memory",
    EXECUTION_LOGS: "Execution Logs",
    ANNOTATION: "Annotation",
  },
};

export default messages;
