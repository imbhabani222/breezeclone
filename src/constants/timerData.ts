const markedData = {
  TIME_LABEL_TWO: `<div class='timer_content_div'>
    <span class='timer_header'>HH</span>
    <div class='timer_borderDiv'>
      <div class='inner_div timer_warning'>%hours%</div>
    </div>
  </div>
  <div class='timer_content_div'>
    <span class='timer_header'>MM</span>
    <div class='timer_borderDiv'>
      <div class='inner_div timer_warning'>%minutes%</div>
    </div>
  </div>
  <div class='timer_content_div'>
    <span class='timer_header'>SS</span>
    <div class='timer_borderDiv'>
      <div class='inner_div timer_warning'>%seconds%</div>
    </div>
  </div>`,
};

export default markedData;
