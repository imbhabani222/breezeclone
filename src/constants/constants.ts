import Cookies from "universal-cookie";
import messages from "./messages";

import { useLocation } from "react-router-dom";
const cookies = new Cookies();
const csrfToken = cookies.get("doselectcsrf");

const constants = {
  HTTP_REQ_DEFAULT_HEADERS: {
    Accept: "application/json",
    "Content-Type": "application/json; charset=UTF-8",
  },
  HTTP_POST_HEADERS: {},
  HTTP_MULTIPART_FORM_HEADERS: {
    "Content-Type": false,
  },
  HTTP_METHODS: {
    POST: "POST",
    GET: "GET",
    DELETE: "DELETE",
    PATCH: "PATCH",
    PUT: "PUT",
  },
  HTTP_REQ_ERROR_MAP: {
    401: { status: 401, message: messages.NOT_AUTHORIZED },
    403: { status: 403, message: messages.ACCESS_DENIED },
    404: { status: 404, message: messages.NO_DATA_AVAILABLE },
  },
  ROUTE: {
    LANDING_PAGE: "/test/:slug/landingPage",
    INSTRUCTION_PAGE: "/test/:slug/instructionPage",
    DOSELECT_INSTRUCTION: "/test/:slug/doSelectInstruction",
    TEST_SECTION_SUBMIT: "/test/:slug/problems",
    FAQ: "/test/:slug/Faqs",
    FEEDBACK: "/test/:slug/Feedback",
    TESTCOMPLETE: "/test/:slug/TestComplete",
    TESTLAYOUT: "/test/:slug/TestLayout",
    RICH_TEXT_EDITOR: "/test/:slug/problems/id",
    FIll_IN_BLANKS: "/tegst/:slug/problems/:id",
    MCQ_PAGE: "/tegst/:slug/problems/:id",
    SWITCH_PATH: "/test/:slug/problems/:id",
  },
  NAME_KEY: "name",
  STATUS_KEY: "status",
  LANGUAGE_KEY: "Language",
  VERSION_KEY: "Version",
  MU_KEY: "Mu",
  TL_KEY: "Tl",
  LA_KEY: "LA",
  DESCRIPTION_KEY: "description",
  DEFAULT_KEY: "default",
  VIDEO_KEY: "video",
  PROGRESSBAR_KEY: "progressbar",
};

export default constants;
