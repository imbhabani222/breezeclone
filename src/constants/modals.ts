import { useState } from "react";
import messages from "./messages";

import useFetch from "./../data/dataBar-async";

import { getDuration, getTimeDiff, stringFiller } from "../data/utils";
import ProgressBarWrapper from "./../components/dataBar/progressBarWrapper/progressBarWrapper";
import { getProblemSelector } from "../selectors/problemSelector";
import { useSelector } from "react-redux";

const {
  FEEDBACK: { SUBMIT_LABEL },
  CANCEL,
  SUBMIT_MOVE_NEXT,
  SUBMIT_END,
  ARE_YOU_SURE,
  REPORT_LABLE,
  PROCEED_TO_NEXT,
  MODAL_CONTENT,
  TYPE_OUTLINE,
  TYPE_FILLED,
  AUTO_NEXT_MODAL,
  PROCEED_TO_FEEDBACK,
  GOING_GREAT,
} = messages;

const GetString = (timeLeft: any, str: string) => {
  const problems = useSelector(getProblemSelector);
  let total: number = 0,
    completed: number = 0;

  if (problems?.length >= 0) {
    problems &&
      problems.forEach((sec: any) => {
        sec.problems.forEach((singleObj: any) => {
          total++;
          if (singleObj.solution !== null) {
            completed++;
          }
        });
      });
  }

  return stringFiller(str, {
    ...timeLeft,
    total: total,
    completed: completed,
  });
};

const GetNextString = (timeLeft: any, str: string) => {
  const problems = useSelector(getProblemSelector);
  let SectionTotal: number = 0,
    sectionCompleted: number = 0;

  if (problems.length >= 0) {
    let current =
      problems &&
      problems?.find((val: any) => {
        return val.status === "CTK";
      });
    current &&
      current?.problems.map((val: any) => {
        if (val.solution !== null) {
          sectionCompleted++;
        }
      });
    SectionTotal = current && current.problems.length;
  }
  // console.log({
  //   ...timeLeft,
  //   total: SectionTotal,
  //   completed: sectionCompleted,
  // });

  return (
    stringFiller(str, {
      ...timeLeft,
      total: SectionTotal,
      completed: sectionCompleted,
    }) || []
  );
};

export const getSubmitConformationModal = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: ARE_YOU_SURE,
    onOk: onOk,
    onCancel: onCancel,
    text: contentType ? GetString(timeLeft, MODAL_CONTENT) : text,
    buttons: contentType
      ? [
          {
            type: TYPE_OUTLINE,
            display: CANCEL,
            onClick: onCancel,
          },
          {
            type: TYPE_FILLED,
            display: SUBMIT_END,
            onClick: onOk,
          },
        ]
      : null,
  };
};

export const getSubmitNextConformationModal = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: ARE_YOU_SURE,
    onOk: onOk,
    onCancel: onCancel,
    text: contentType ? GetNextString(timeLeft, MODAL_CONTENT) : text,
    buttons: contentType
      ? [
          {
            type: TYPE_OUTLINE,
            display: CANCEL,
            onClick: onCancel,
          },
          {
            type: TYPE_FILLED,
            display: SUBMIT_MOVE_NEXT,
            onClick: onOk,
          },
        ]
      : null,
  };
};
export const autoRedirectModalToNext = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: "",
    onOk: onOk,
    onCancel: onCancel,
    text: contentType ? text : GetString(timeLeft, AUTO_NEXT_MODAL),
    buttons: contentType
      ? [
          {
            type: TYPE_FILLED,
            display: PROCEED_TO_FEEDBACK,
            onClick: onOk,
          },
        ]
      : null,
  };
};
export const autoRedirectToSection = ({
  onOk,
  onCancel,
  timeLeft,
  contentType,
  text,
}: any) => {
  return {
    title: "",
    onOk: onOk,
    onCancel: onCancel,
    text: text,
    buttons: contentType
      ? [
          {
            type: TYPE_FILLED,
            display: "Submit and move to next section",
            onClick: onOk,
          },
        ]
      : null,
  };
};
