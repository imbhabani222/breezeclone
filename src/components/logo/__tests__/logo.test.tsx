import React from "react";
import { BrowserRouter, useLocation } from "react-router-dom";
import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import Logo from "../logo";
import messages from "../../../constants/messages";
import constants from "../../../constants/constants";
const { LOGO } = messages;
const {
  ROUTE: { LANDING_PAGE },
} = constants;

describe("Logo", () => {
  function renderLogo() {
    return render(
      <BrowserRouter>
        <Logo />
      </BrowserRouter>
    );
  }
  let location = useLocation();
  afterEach(cleanup);
  test("Render without Crashing", () => {
    renderLogo();
  });
  test("Match snapshot logo component", () => {
    const { asFragment } = renderLogo();
    expect(asFragment()).toMatchSnapshot();
  });

  test("Display Logo", async () => {
    renderLogo();
    expect(screen.getByTitle(LOGO)).toBeInTheDocument();
  });
  //Doubt
  test("click on logo icon", async () => {
    // global.window = { location: { pathname: "/test/sample" } };
    renderLogo();
    expect(screen.getByTitle(LOGO)).toBeTruthy();
    fireEvent.click(screen.getByTitle(LOGO));
    expect(screen.getByTitle(LOGO)).toHaveBeenCalledTimes(1);
    // renderLogo();
    // const Click = screen.getByTitle(LOGO);
    // fireEvent(
    //   Click,
    //   new MouseEvent("click", {
    //     bubbles: true,
    //     cancelable: true,
    //   })
    // );
    // expect(Click).toHaveBeenCalledTimes(0);
  });
});
