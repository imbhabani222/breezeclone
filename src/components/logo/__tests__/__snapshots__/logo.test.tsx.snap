// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Match snapshot logo component 1`] = `
<DocumentFragment>
  <span
    class="anticon logo"
    role="img"
    tabindex="-1"
    title="do select logo"
  >
    <svg
      aria-hidden="true"
      class=""
      fill="currentColor"
      focusable="false"
      height="1em"
      width="1em"
    >
      logo.svg
    </svg>
  </span>
</DocumentFragment>
`;
